### PR TITLE
Mcp 181 jf client cleanup

### PR DESCRIPTION
**Pull request by jmfadeley**
### Related tickets:
MCP-179, MCP-180, MCP-181.

### What's changing? Any breaking changes?
Nothing. Added tests for the data service, changed function names per Grant's request, added a working `Dockerfile`. 

### Manual test cases?
Run `yarn test` and confirm it's all working. Then run `docker build -t mcp-data-client .` followed by `docker run -p 3000:80 -e REACT_APP_MCP_DATA_MAPPING_UTIL=http://localhost:5000/ mcp-data-client` to run a docker container of the front end. If you set your port for the backend to something else, obviously adjust the port accordingly. 

### Any additional context/background?
-

### Screenshots (if appropriate):
-

### CODE REVIEW

File: src/services/data/DataService.test.tsx [hunk 1]
```diff
@@ -1,12 +1,76 @@
...
+              json: "Yes (enum)",
+              message: "values do not match"
+            }
+          ]
+        }
+      };
+
+      const rs = new Stream.Readable();
+      rs.push(JSON.stringify(body));
+      rs.push(null);
```
Comment (TylerNoblett) [inline src/services/data/DataService.test.tsx:49]: What is the purpose of `rs.push(null);`?

**Review by TylerNoblett — CHANGES_REQUESTED** [review]
Everything looks good to me! The only thing I noticed (and you can include this in another ticket if you'd like), is that Eric added an error type with a PR he submitted a few days ago in mapping utils that currently has no heading. I can send you a screenshot and a doc to use to see the error, but here's a look at what the error looks like/it's name:
```
                {
                    "error_type": "mark_for_complexity_is_unchecked",
                    "message": "'Mark for Complexity' should not be blank if 'Complete Data Source' is 'N'",
                    "object_id": f"{json_object_id}",
                }
```

File: src/services/data/DataService.test.tsx (hunk 1 above)
Comment (jmfadeley) [reply src/services/data/DataService.test.tsx:49]: From what I read, I believe it sees `null` as the end for when to stop reading the stream. https://stackoverflow.com/questions/38316821/why-can-i-push-into-a-readable-stream

**Review by TylerNoblett — APPROVED** [review]
new error type is present and tests pass so it's good to go!

**Comment by jmfadeley** [comment]
@TylerNoblett pushed new error type and corresponding tests.

**Pull request merged**